crates/rpc/src/cairo/ext_py/ser.rs

```diff
@@ -339,124 +338,143 @@ impl TryFrom<BlockId> for BlockHashNumberOrLatest {
+            let result = serde_json::to_string(&StorageUpdates(None)).unwrap();
+            assert_eq!(result, "{}");
+        }
+
+        #[test]
+        fn with_updates() {
+            use pathfinder_common::{StorageAddress, StorageValue};
+
+            let expected = r#"{
+                "0x7c38021eb1f890c5d572125302fe4a0d2f79d38b018d68a9fcd102145d4e451":[{"key":"0x5","value":"0xabc"}],
+                "0x1":[{"key":"0x123","value":"0xdef"}]
+            }"#;
+            // Use Value so we don't run into ordering issues.
+            let expected: serde_json::Value = serde_json::from_str(expected).unwrap();
+
+            let update = StateUpdate::default()
+                .with_storage_update(
+                    ContractAddress::new_or_panic(felt!(
+                        "07c38021eb1f890c5d572125302fe4a0d2f79d38b018d68a9fcd102145d4e451"
+                    )),
+                    StorageAddress::new_or_panic(felt!("0x5")),
+                    StorageValue(felt!("0xabc")),
+                )
+                .with_system_storage_update(
+                    ContractAddress::ONE,
+                    StorageAddress::new_or_panic(felt!("0x123")),
+                    StorageValue(felt!("0xdef")),
+                );
+            let s = serde_json::to_value(StorageUpdates(Some(&update))).unwrap();
+            assert_eq!(expected, s);
+        }
     }
 
-    #[test]
-    fn serialize_some_deployed_contracts() {
-        use super::{DeployedContract, DeployedContractsWrapper};
+    mod deployed_contracts {
+        use super::*;
         use pathfinder_common::ClassHash;
 
-        let expected = r#"[{"address":"0x7c38021eb1f890c5d572125302fe4a0d2f79d38b018d68a9fcd102145d4e451","contract_hash":"0x10455c752b86932ce552f2b0fe81a880746649b9aee7e0d842bf3f52378f9f8"}]"#;
-        let contracts = vec![DeployedContract {
-            address: ContractAddress::new_or_panic(felt!(
-                "07c38021eb1f890c5d572125302fe4a0d2f79d38b018d68a9fcd102145d4e451"
-            )),
-            class_hash: ClassHash(felt!(
-                "010455c752b86932ce552f2b0fe81a880746649b9aee7e0d842bf3f52378f9f8"
-            )),
-        }];
-        let s = serde_json::to_string(&DeployedContractsWrapper(Some(&contracts))).unwrap();
-        assert_eq!(expected, s);
-
-        // this again could be null or []; it doesn't really have any meaning over at python side,
-        // nor is it coupled to the updated contracts value in any way
-        let s = serde_json::to_string(&DeployedContractsWrapper(Some(&[]))).unwrap();
-        assert_eq!("[]", s);
-    }
+        #[test]
+        fn none() {
+            let result = serde_json::to_string(&DeployedContracts(None)).unwrap();
+            assert_eq!(result, "[]");
+        }
 
-    /// It is important that this is not `[]` or `{}`, see [`serialize_none_updates`].
-    #[test]
-    fn serialize_none_deployed_contracts() {
-        use super::DeployedContractsWrapper;
+        #[test]
```

Comment:
The same here -- these are exactly the same now, I'd only keep the `none` variant.